### ISSUE TITLE
included js via layout update in cms pages is broken

### ISSUE THREAD

**Issue by ioweb-gr** [magento/magento2#26733]
### Preconditions (*)
Magento 2.3.4

### Steps to reproduce (*)
1. Create a simple CMS page 
2. In your module add a js file app\code\Ioweb\Customizer\view\frontend\web\js\myfile.js

```
require(['jquery'], function ($) {
    $(document).ready(function () {
        console.log('myfile');
    });
});

```

3. In a CMS page include in the layout update section the following
```
    <head>
        <script src="Ioweb_Customizer::js/myfile.js" />
    </head>

```
Open the cms page and you get an error `TypeError: require is not a function`

If you do add the same rule in layout `default.xml` for example it works OK

```
    <head>
        <script src="Ioweb_Customizer::js/myfile.js" />
    </head>
```

### Expected result (*)
1. Layout updates allow us to add js files like normal layout files

### Actual result (*)
1. JS file included through layout updates in CMS pages cannot use require 

I'm assuming this affect categories and other objects too but haven't tested those.

**Comment by mrtuvn** [comment]
Add js in cms in not best practice you should avoid that
In fact in next update magento introduce new way to update layout via cms
You can take a look here
@ioweb-gr you can take a look at this
https://devdocs.magento.com/guides/v2.3/frontend-dev-guide/layouts/xml-manage.html#create-cms-pageproductcategory-specific-layouts

**Comment by ioweb-gr** [comment]
@mrtuvn I see your point. However including a script in all pages when you only need it for a specific one is not good either. For now I have to add it in default.xml and then inside the javascript check if I'm in the correct page. 

I do like the new options you pointed out and will try them out when we update the eshop. 

However I still think that if the option is provided to store a layout update in the database, then it should behave like a normal layout update. 

Otherwise why is it there in the first place? As it is now  the developer would have to guess which directive works and which doesn't.

**Comment by mrtuvn** [comment]
Magento only introduce new way but i don't see how to migrate old data custom layout yet
If you have alot of category and each category have layout updates. It's will a lot of works to migrate in new 2.3.4

**Comment by ioweb-gr** [comment]
For all my older websites I loaded the scripts globally unfortunately and now they handle it based on whether they found the needed element in the current page. Obviously this was not the best way so in this new website I wanted to try this option. 

Still for some use cases, the end user is not necessarily allowed to use FTP to upload new files but there's a very good way in the layout updates from the backend to perform such actions so I don't see why there's no valid reason to fix it.

**Comment by ioweb-gr** [comment]
@mrtuvn I read a few articles regarding this update for custom layouts and it seems that people who upgraded mention that the old directives added via the textarea box in backend are still in effect, it's just admins that lost the ability to edit them. Several mention that upgrading didn't affect their store. 

However I can verify from my own experience that installing sample data on a lot of themes I tried (over 10) became impossible now, because it's impossible to save the layout updates in cms pages programmatically and it throws an error. Most commercial themes use custom layout updates for the home page at least.

**Comment by engcom-Golf** [comment]
HI @ioweb-gr thank you for you report, issue is not reproducible in 2.4-develop branch, this may be related to some custom modules, or syntax in your js file. please see my attachments
![DeepinScreenshot_select-area_20200303145553](https://user-images.githubusercontent.com/51681547/75777649-fe1f6200-5d5e-11ea-8036-2cbb24f598ca.png)
![DeepinScreenshot_select-area_20200303145401](https://user-images.githubusercontent.com/51681547/75777665-07103380-5d5f-11ea-8fc8-7797d7873797.png)
![DeepinScreenshot_select-area_20200303145522](https://user-images.githubusercontent.com/51681547/75777668-07a8ca00-5d5f-11ea-9d5f-dc36f8d558ba.png)
![DeepinScreenshot_select-area_20200303145539](https://user-images.githubusercontent.com/51681547/75777669-08416080-5d5f-11ea-96b2-def61d691056.png)

**Comment by ioweb-gr** [comment]
@engcom-Golf Actually if you read the previous posts, it seems this issue has become impossible to replicate now directly in 2.4 and 2.3.4 because the textarea for writing layout  updates is disabled for the admin and needs to be done via FTP and adding new XML files there. Unfortunately I haven't tried the new way yet to see if the issue persists but should I encounter it, it will be food for a new issue. 

This was a BC Breaking change introduced since 2.3.4 which disabled the users from having this ability. However the existing rules according to various mentions are still active in the background and therefore should present the same issue. I'm thinking this as a BC break because after you have put a script in previous magento versions and upgrade you don't have a way to clear it up. 

If you really want to replicate this behaviour you'd have to install 2.3.3, write the layout  update and test there. The problem will definitely occur. Afterwards, upgrade to 2.4 and see that the same script that was included in 2.3.3 will still be in effect and maybe still broken. 

I'm not sure whether it has any point fixing it now though, but it affects users running 2.3.1->2.3.3 from what I tested. It could be present in people in 2.2.x branch as well but I don't have an installation for that version to test.

**Comment by engcom-Golf** [comment]
@ioweb-gr for textarea i think that this is expected behavior, thank you for investigate, i will take a look at this one more time.
